analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MagicNumberShouldNotBeUsed.cs

```diff
@@ -14,18 +34,31 @@ public static double FooProp
             get { return 2; }
         }
 
-        public ValidUseCases()
+        public ValidUseCases(string s, FooBar foo)
         {
             int i1 = 0;
             int i2 = -1;
             int i3 = 1;
             int i4 = 42;
+            var list = new List<string>(42); // Compliant, class constructor
+            var bigInteger = new IntPtr(1612342); // Compliant, class constructor
+            var g = new Guid(0xA, 0xB, 0xC, new Byte[] { 0, 1, 2, 3, 4, 5, 6, 7 }); // Compliant, class constructor
+            long[] values1 = { 30L, 40L }; // Compliant, array initialisation
+            int[] values2 = new int[] { 100, 200 }; // Compliant, array initialisation
```

Comment:
add test with `Foo(new int[] { 100, 200 })`

Reply:
I agree. Collection initialization should be Noncompliant unless in UTs.